packages/neo-one-editor/src/manager/FetchQueue.ts

```diff
@@ -0,0 +1,52 @@
+import fetch from 'cross-fetch';
+
+export interface QueueData<T> {
+  readonly url: string;
+  readonly handleResponse: (response: Response) => Promise<T>;
+  readonly resolve: (result: T) => void;
+  readonly reject: (error: Error) => void;
+}
+
+interface FetchQueueOptions {
+  readonly fetchConcurrency: number;
+}
+
+export class FetchQueue<T> {
+  private readonly fetchConcurrency: number;
+  private readonly mutableQueue: Array<QueueData<T>>;
+  private mutableRunning: number;
+
+  public constructor(options?: FetchQueueOptions) {
+    this.fetchConcurrency = options === undefined ? 10 : options.fetchConcurrency;
+    this.mutableQueue = [];
+    this.mutableRunning = 0;
+  }
+
+  public push(queueData: QueueData<T>) {
+    this.mutableQueue.push(queueData);
+  }
+
+  public advance() {
```

Comment:
This should be `private`, callers shoudn't have to call `advance` or even know about it.